pkgpanda/__init__.py

```diff
@@ -42,14 +41,27 @@
 version_regex = "^[a-zA-Z0-9@_+:.]+$"
 username_regex = "^dcos_[a-z0-9_]+$"
 
+dcos_sysctl_config = '/etc/sysctl.d/dcos.conf'
+
+
+def preserve_sysctl(sysctl_config_file, preserved_sysctl_map):
+    sysctl_setting = "\n".join(["{key} = {value}".format(key=key, value=value)
+                                for key, value in preserved_sysctl_map.items()])
+    write_string(sysctl_config_file, sysctl_setting)
+
+
+def activate_sysctl(sysctl_config_file):
+    # NOTE: '-e' flag ignores errors on invalid sysctl flag.
```

Comment:
`-e` ignores the error on unknown keys to `sysctl`.  Given that I migrated sysctl setting from `build` file to buildinfo.json, This was being exercised in CI and CI system itself does not support some flags.

[Team City Job Details](https://teamcity.mesosphere.io/viewLog.html?buildId=337779&buildTypeId=DcosIo_Dcos_BuildPr)

```
[17:30:57][Step 1/1] sysctl: cannot stat /proc/sys/net/netfilter/ip_conntrack_tcp_be_liberal: No such file or directory
[17:30:57][Step 1/1] sysctl: cannot stat /proc/sys/net/ipv4/netfilter/ip_conntrack_tcp_be_liberal: No such file or directory
[17:30:57][Step 1/1] Traceback (most recent call last):
[17:30:57][Step 1/1]   File "/teamcity/work/19ad0dff41587ee6/build/env/bin/release", line 11, in <module>
[17:30:57][Step 1/1]     sys.exit(main())
[17:30:57][Step 1/1]   File "/teamcity/work/19ad0dff41587ee6/build/env/lib/python3.4/site-packages/release/__init__.py", line 746, in main
[17:30:57][Step 1/1]     release_manager.create('testing', options.channel, options.tag)
[17:30:57][Step 1/1]   File "/teamcity/work/19ad0dff41587ee6/build/env/lib/python3.4/site-packages/release/__init__.py", line 639, in create
[17:30:57][Step 1/1]     metadata = make_stable_artifacts(self.__config['options']['cloudformation_s3_url'] + '/' + repository_path)
[17:30:57][Step 1/1]   File "/teamcity/work/19ad0dff41587ee6/build/env/lib/python3.4/site-packages/release/__init__.py", line 284, in make_stable_artifacts
[17:30:57][Step 1/1]     all_bootstraps = do_build_packages(cache_repository_url)
[17:30:57][Step 1/1]   File "/teamcity/work/19ad0dff41587ee6/build/env/lib/python3.4/site-packages/release/__init__.py", line 465, in do_build_packages
[17:30:57][Step 1/1]     return get_build()
[17:30:57][Step 1/1]   File "/teamcity/work/19ad0dff41587ee6/build/env/lib/python3.4/site-packages/release/__init__.py", line 456, in get_build
[17:30:57][Step 1/1]     result = pkgpanda.build.build_tree(package_store, True, None)
[17:30:57][Step 1/1]   File "/teamcity/work/19ad0dff41587ee6/build/env/lib/python3.4/site-packages/pkgpanda/build/__init__.py", line 665, in build_tree
[17:30:57][Step 1/1]     results[variant] = make_bootstrap(variant)
[17:30:57][Step 1/1]   File "/teamcity/work/19ad0dff41587ee6/build/env/lib/python3.4/site-packages/pkgpanda/build/__init__.py", line 658, in make_bootstrap
[17:30:57][Step 1/1]     variant)
[17:30:57][Step 1/1]   File "/teamcity/work/19ad0dff41587ee6/build/env/lib/python3.4/site-packages/pkgpanda/build/__init__.py", line 427, in make_bootstrap_tarball
[17:30:57][Step 1/1]     install.activate(repository.load_packages(pkg_ids))
[17:30:57][Step 1/1]   File "/teamcity/work/19ad0dff41587ee6/build/env/lib/python3.4/site-packages/pkgpanda/__init__.py", line 745, in activate
[17:30:57][Step 1/1]     self.swap_active(".new")
[17:30:57][Step 1/1]   File "/teamcity/work/19ad0dff41587ee6/build/env/lib/python3.4/site-packages/pkgpanda/__init__.py", line 847, in swap_active
[17:30:57][Step 1/1]     activate_sysctl(self.sysctl_config)
[17:30:57][Step 1/1]   File "/teamcity/work/19ad0dff41587ee6/build/env/lib/python3.4/site-packages/pkgpanda/__init__.py", line 54, in activate_sysctl
[17:30:57][Step 1/1]     check_call(["sysctl", "-q", "-p", sysctl_config_file])
[17:30:57][Step 1/1]   File "/usr/lib/python3.4/subprocess.py", line 561, in check_call
[17:30:57][Step 1/1]     raise CalledProcessError(retcode, cmd)
[17:30:57][Step 1/1] subprocess.CalledProcessError: Command '['sysctl', '-q', '-p', '/etc/sysctl.d/dcos.conf']' returned non-zero exit status 255
[17:30:57][Step 1/1] Process exited with code 1
[17:30:57][Step 1/1] Step Command Line failed
[17:30:57]Publishing internal artifacts
[17:30:57]Publishing artifacts (5s)

```

Our CI system does not support the flags that the minuteman expects and these are actally set when the package is installed. 

Btw, stubbing out is done in tests. This sysctl flag is a package requirement, so either they need to be exercised by the tests on proper hardware or if they are incorrect, they should be removed. 

How should we go about this?